### ISSUE TITLE
Load data error 

### ISSUE THREAD

**Issue by Donxiaohao** [dolthub/dolt#8307]
mysql version 14.14
dolt version 1.42.11
![image](https://github.com/user-attachments/assets/bbc8a900-0288-4da7-9bb8-2b7284f6d34c)
When I follow the official recommendation to batch import and use the load data command,I encountered the error shown above.
![image](https://github.com/user-attachments/assets/4a85220c-1f67-469f-96c3-7395f6a1053e)
The same command is executed normally in mysql
![image](https://github.com/user-attachments/assets/955ac84d-23b2-4333-94b9-047bd365bf8c)
Looking at the error message, the problem lies in the set, but the official website syntax description is supported.
I think this is a bug，

**Comment by timsehn** [comment]
This looks like we don't support SET syntax in our LOAD DATA. @jycor will look into this today. It may be easy to supportor it may take a few days.

**Comment by zachmu** [comment]
@jycor can you summarize what's left to do on this one?

**Comment by jycor** [comment]
The two linked PRs (https://github.com/dolthub/go-mysql-server/pull/2653 and https://github.com/dolthub/vitess/pull/362) add support for the `SET` clase in `LOAD DATA` statements, so we're able to assign column values during a `LOAD DATA`.

However, we are missing support for assigning user variables within a `LOAD DATA` query.
For example:
```sql
LOAD DATA INFILE <file_path> INTO TABLE t (i, j, @k) set k = concat('123', @k);
```

I'm currently working adding this. 
Hopefully, will have something by the end of the week.

**Comment by jycor** [comment]
Hey @Donxiaohao, thanks for making this issue!
We've managed to support the features requested for `LOAD DATA`.

The fix is making its way to main, and we will have a release out for you later this week!
Please let us know if run into any more issues.

**Comment by Donxiaohao** [comment]
@jycor Thank you so much ， I will try
